pkg/kubelet/kubelet_test.go

```diff
@@ -1908,6 +1909,104 @@ func TestDoesNotDeletePodDirsIfContainerIsRunning(t *testing.T) {
 	syncAndVerifyPodDir(t, testKubelet, pods, []*v1.Pod{apiPod}, false)
 }
 
+func DisabledTestFastStatusUpdateOnce(t *testing.T) {
+	tests := []struct {
+		name          string
+		exitOnSuccess bool
+	}{
+		{
+			name:          "exit on timeout",
+			exitOnSuccess: false,
+		},
+		{
+			name:          "exit on success",
+			exitOnSuccess: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+			defer testKubelet.Cleanup()
+			kubelet := testKubelet.kubelet
+			// Ensure we capture actions on the heartbeat client only.
+			// We don't set it to nil or GetNode() doesn't read from nodeLister.
+			kubelet.kubeClient = &fake.Clientset{}
+			kubeClient := testKubelet.fakeKubeClient
+
+			node := &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: string(kubelet.nodeName),
+				},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:    v1.NodeReady,
+							Status:  v1.ConditionFalse,
+							Reason:  "NotReady",
+							Message: "Node not ready",
+						},
+					},
+				},
+			}
+
+			nodeLister := testNodeLister{[]*v1.Node{node.DeepCopy()}}
+
+			// Testing for the failure case: we want the first syncNodeStatus() to fail.
+			// With UpdateNodeStatusFromLister, it will try `nodeStatusUpdateRetry` patch actions before returning with error
+			// (one for lister GET, and four for apiserver GET for nodeStatusUpdateRetry = 5).
+			patchCount := 0
+			kubeClient.AddReactor("patch", "nodes", func(action core.Action) (bool, runtime.Object, error) {
+				patchCount++
+				if patchCount <= nodeStatusUpdateRetry {
+					return true, nil, fmt.Errorf("try again")
+				}
+				if tc.exitOnSuccess {
+					nodeLister.nodes[0].Status.Conditions[0].Status = v1.ConditionTrue
+				} else {
+					testKubelet.fakeClock.Step(2 * nodeReadyGracePeriod)
+				}
+				return false, nil, nil
+			})
+
+			kubelet.nodeLister = nodeLister
+
+			start := time.Now()
+			kubelet.fastStatusUpdateOnce()
+			assert.Less(t, time.Since(start), nodeReadyGracePeriod)
+
+			actions := kubeClient.Actions()
+			actionIsPatch := []bool{true}
+			for i := 1; i < nodeStatusUpdateRetry; i++ {
+				actionIsPatch = append(actionIsPatch, false, true)
+			}
+			actionIsPatch = append(actionIsPatch, true)
+			require.Len(t, actions, len(actionIsPatch))
+
+			for i, action := range actions {
```

Comment:
so, this goes all over all the actions performed, and check that the patches set the condition to ready and if is not a patch is a Get

Reply:
I'll probably need to rewrite this part.

Reply:
tests updated.